cscs-checks/apps/vasp/vasp_check.py

```diff
@@ -7,130 +7,124 @@
+        valid_systems = {
+            6: ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                'eiger:mc', 'pilatus:mc'],
+            16: ['daint:gpu', 'daint:mc', 'eiger:mc']
         }
-        self.reference = references[self.scale]
 
-    @run_before('run')
-    def set_task_distribution(self):
-        self.job.options = ['--distribution=block:block']
-
-    @run_before('run')
-    def set_cpu_binding(self):
-        self.job.launcher.options = ['--cpu-bind=cores']
+        try:
+            self.valid_systems = valid_systems[self.num_nodes]
+        except KeyError:
+            self.valid_systems = []
 
+        # setup programming environment filter
+        if self.current_system.name in ['eiger', 'pilatus']:
+            self.valid_prog_environs = ['cpeIntel']
+        else:
+            self.valid_prog_environs = ['builtin']
 
-@rfm.simple_test
-class VASPGpuCheck(VASPCheck):
-    valid_systems = ['daint:gpu']
-    num_gpus_per_node = 1
-    num_tasks_per_node = 1
-    num_cpus_per_task = 12
-    use_multithreading = False
 
-    @run_after('init')
-    def setup_test(self):
-        self.descr = (f'VASP GPU check (version: {self.scale}, '
-                      f'{self.variant})')
-        if self.scale == 'small':
-            self.valid_systems += ['dom:gpu']
-            self.num_tasks = 6
-        else:
-            self.num_tasks = 16
-
-    @run_before('performance')
-    def set_reference(self):
-        references = {
-            'small': {
-                'dom:gpu': {'time': (65.0, None, 0.15, 's')},
-                'daint:gpu': {'time': (65.0, None, 0.15, 's')}
-            },
-            'large': {
-                'daint:gpu': {'time': (55.0, None, 0.15, 's')}
+    @run_before('run')
+    def setup_run(self):
+        # set auto-detected architecture
+        proc = self.current_partition.processor
+        arch = proc.arch
+        # set architecture for GPU partition (no auto-detection)
+        if self.current_partition.fullname in ('daint:gpu', 'dom:gpu'):
+            arch = 'sm_60'
+
+        try:
+            found = self.references[self.num_nodes][arch]
+        except KeyError:
+            self.skip(f'Configuration with {self.num_nodes} node(s) '
+                      f'is not supported on {arch!r}')
+
+        # custom settings for each architecture
+        if arch == 'sm_60':
+            self.num_tasks_per_node = 1
+        elif arch == 'broadwell':
+            self.num_tasks_per_node = 2
+        elif arch == 'zen2':
+            self.num_tasks_per_node = 4
+            self.variables = {
+                'MPICH_OFI_STARTUP_CONNECT': '1'
             }
+
+        # common setup for every architecture
+        self.job.launcher.options = ['--cpu-bind=cores']
+        self.job.options = ['--distribution=block:block']
+        self.num_cpus_per_task = int(proc.num_cores / self.num_tasks_per_node)
+        self.num_tasks = self.num_nodes * self.num_tasks_per_node
+        self.variables = {
```

Comment:
Here you override the `variables` for the `zen2` nodes.

Reply:
Modified with `self.variables += ...`, thanks